asmcomp/selectgen.ml

```diff
@@ -690,6 +673,15 @@ method emit_expr (env:environment) exp =
         None -> None
       | Some r1 -> self#emit_expr (self#bind_let env v r1) e2
       end
+  | Clet_mut(v, k, e1, e2) ->
+      begin match self#emit_expr env e1 with
+        None -> None
+      | Some r1 ->
+        let rv = Reg.createv k in
+        name_regs v rv;
+        self#insert_moves env r1 rv;
+        self#emit_expr (env_add v rv env) e2
+      end
```

Comment:
This is quite similar to bind_let, I think it would be more readable if you managed to merge them together.

Reply:
I agree about this version of the code, but once I made the Cassign assertion change they only share two lines, so I don't think there's any improvement to be made by merging them.